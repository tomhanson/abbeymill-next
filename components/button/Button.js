// @flow strict
import * as React from 'react';
import { StyledButton, StyledLink } from './Button.styles';

type Props = {
  primary: ?boolean,
  secondary: ?boolean,
  href: ?string,
  children: React.Node,
};

function Button(props: Props) {
  return (
    <StyledLink href={props.href} primary={props.primary} secondary={props.secondary}>
      {props.children}
    </StyledLink>
  );
}

export default Button;
