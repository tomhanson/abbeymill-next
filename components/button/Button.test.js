import { render } from 'react-testing-library';
import Button from './Button';

const defaultProps = {
  url: '/test/test',
};

describe('Button component', () => {
  test('the component loads correctly with only default props', () => {
    const component = render(<Button {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  test('the component loads correctly with primary set to true', () => {
    const component = render(<Button {...defaultProps} primary />);
    expect(component).toMatchSnapshot();
  });
  test('the component loads correctly with secondary set to true', () => {
    const component = render(<Button {...defaultProps} secondary />);
    expect(component).toMatchSnapshot();
  });
});
