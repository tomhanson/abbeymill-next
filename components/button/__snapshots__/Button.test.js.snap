// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component the component loads correctly with only default props 1`] = `
Object {
  "baseElement": <html>
    <head>
      <style
        data-styled-components=""
      >
        
        
/* sc-component-id: sc-bdVaJa */

        
/* sc-component-id: sc-bwzfXH */
.sc-bwzfXH {} .cxPwlj{display:inline-block;margin-bottom:0;font-weight:400;text-align:center;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;line-height:1.428571429 !important;border-radius:4px;background-image:none;border:1px solid transparent;white-space:nowrap;-webkit-text-decoration:none;text-decoration:none;font-size:14px;padding:10px;padding:1rem;color:#ffffff;-webkit-appearance:none;border:1px solid transparent;background-color:transparent;} .cxPwlj:hover,.cxPwlj:active,.cxPwlj:focus{color:#ffffff;}
      </style>
    </head>
    <body>
      <div>
        <a
          class="sc-bwzfXH cxPwlj"
        />
      </div>
    </body>
  </html>,
  "container": <div>
    <a
      class="sc-bwzfXH cxPwlj"
    />
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component the component loads correctly with primary set to true 1`] = `
Object {
  "baseElement": <html>
    <head>
      <style
        data-styled-components=""
      >
        
        
/* sc-component-id: sc-bdVaJa */

        
/* sc-component-id: sc-bwzfXH */
.sc-bwzfXH {} .cxPwlj{display:inline-block;margin-bottom:0;font-weight:400;text-align:center;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;line-height:1.428571429 !important;border-radius:4px;background-image:none;border:1px solid transparent;white-space:nowrap;-webkit-text-decoration:none;text-decoration:none;font-size:14px;padding:10px;padding:1rem;color:#ffffff;-webkit-appearance:none;border:1px solid transparent;background-color:transparent;} .cxPwlj:hover,.cxPwlj:active,.cxPwlj:focus{color:#ffffff;}.dhDIbt{display:inline-block;margin-bottom:0;font-weight:400;text-align:center;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;line-height:1.428571429 !important;border-radius:4px;background-image:none;border:1px solid transparent;white-space:nowrap;-webkit-text-decoration:none;text-decoration:none;font-size:14px;padding:10px;padding:1rem;color:#ffffff;-webkit-appearance:none;border:1px solid transparent;background-color:#408353;-webkit-transition:0.3s background-color;transition:0.3s background-color;} .dhDIbt:hover,.dhDIbt:active,.dhDIbt:focus{background-color:darken($brand-four,10%);color:#ffffff;}
      </style>
    </head>
    <body>
      <div>
        <a
          class="sc-bwzfXH cxPwlj"
        />
      </div>
      <div>
        <a
          class="sc-bwzfXH dhDIbt"
        />
      </div>
    </body>
  </html>,
  "container": <div>
    <a
      class="sc-bwzfXH dhDIbt"
    />
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component the component loads correctly with secondary set to true 1`] = `
Object {
  "baseElement": <html>
    <head>
      <style
        data-styled-components=""
      >
        
        
/* sc-component-id: sc-bdVaJa */

        
/* sc-component-id: sc-bwzfXH */
.sc-bwzfXH {} .cxPwlj{display:inline-block;margin-bottom:0;font-weight:400;text-align:center;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;line-height:1.428571429 !important;border-radius:4px;background-image:none;border:1px solid transparent;white-space:nowrap;-webkit-text-decoration:none;text-decoration:none;font-size:14px;padding:10px;padding:1rem;color:#ffffff;-webkit-appearance:none;border:1px solid transparent;background-color:transparent;} .cxPwlj:hover,.cxPwlj:active,.cxPwlj:focus{color:#ffffff;}.dhDIbt{display:inline-block;margin-bottom:0;font-weight:400;text-align:center;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;line-height:1.428571429 !important;border-radius:4px;background-image:none;border:1px solid transparent;white-space:nowrap;-webkit-text-decoration:none;text-decoration:none;font-size:14px;padding:10px;padding:1rem;color:#ffffff;-webkit-appearance:none;border:1px solid transparent;background-color:#408353;-webkit-transition:0.3s background-color;transition:0.3s background-color;} .dhDIbt:hover,.dhDIbt:active,.dhDIbt:focus{background-color:darken($brand-four,10%);color:#ffffff;}.DoNiP{display:inline-block;margin-bottom:0;font-weight:400;text-align:center;vertical-align:middle;-ms-touch-action:manipulation;touch-action:manipulation;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;line-height:1.428571429 !important;border-radius:4px;background-image:none;border:1px solid transparent;white-space:nowrap;-webkit-text-decoration:none;text-decoration:none;font-size:14px;padding:10px;padding:1rem;color:#ffffff;-webkit-appearance:none;border:1px solid #ffffff;background-color:transparent;} .DoNiP:hover,.DoNiP:active,.DoNiP:focus{color:#ffffff;}
      </style>
    </head>
    <body>
      <div>
        <a
          class="sc-bwzfXH cxPwlj"
        />
      </div>
      <div>
        <a
          class="sc-bwzfXH dhDIbt"
        />
      </div>
      <div>
        <a
          class="sc-bwzfXH DoNiP"
        />
      </div>
    </body>
  </html>,
  "container": <div>
    <a
      class="sc-bwzfXH DoNiP"
    />
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
